Mathlib/Algebra/Group/Opposite.lean

```diff
@@ -93,24 +93,28 @@ We also generate additive structures on `αᵃᵒᵖ` using `to_additive`
 
 
 @[to_additive]
-instance semigroup [Semigroup α] : Semigroup αᵐᵒᵖ :=
-  { MulOpposite.mul α with
-    mul_assoc := fun x y z => unop_injective <| Eq.symm <| mul_assoc (unop z) (unop y) (unop x) }
+instance isRightCancelMul [Mul α] [IsLeftCancelMul α] : IsRightCancelMul αᵐᵒᵖ where
+  mul_right_cancel := fun _ _ _ => (unop_injective <| mul_left_cancel <| op_injective ·)
```

Comment:
The point in `where` is that you can write
```suggestion
  mul_right_cancel _ _ _ := (unop_injective <| mul_left_cancel <| op_injective ·)
```

Reply:
Yes, I only noticed it later...  I force pushed a version without the arrow!